### PR TITLE
Fix funclet begin assert

### PR DESCRIPTION
**Pull request by BruceForstall**

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Comment by BruceForstall** [comment]
@briansull PTAL
cc @dotnet/jit-contrib

File: src/jit/codegenarm64.cpp [hunk 1]
```diff
@@ -906,7 +906,7 @@ void CodeGen::genFuncletProlog(BasicBlock* block)
 #endif
 
     assert(block != NULL);
-    assert(block->bbFlags && BBF_FUNCLET_BEG);
+    assert(block->bbFlags & BBF_FUNCLET_BEG);
```
Comment (pgavlin) [inline src/jit/codegenarm64.cpp]: Nit: would you mind fixing up the `NULL` above and changing the assert you've corrected to `assert((block->bbFlags & BBF_FUNCLET_BEG) != 0)`?

**Comment by briansull** [comment]
LGTM

**Pull request merged**